### ISSUE TITLE
Enhancement: Custom user fields

### ISSUE THREAD

**Issue by pmfx** [modxcms/evolution#803]
It would be great to be able to create custom fields for Manager/Web users. This thing is always missing when developing site with web users. 

Admin should be able to easily add custom user fields like he is able to create TVs for templates now. Ideally, fields would be assigned to user groups.

In Revolution there is an option to add custom fields but it is not working well, because created fields applies only to the one user. 

In Evolution I would like it be global for all users or organized by user groups.

I think it should be a priority feature in next release after 1.1.1

**Comment by Nicola1971** [comment]
100% agree

**Comment by bossloper** [comment]
+1 for this.

This should simplify the need for the likes of WebLoginPE which is complex and old. The addition of a templatable snippet for frontend members to edit their profile and maybe a 2016 refresh of weblogin if needed and that would about cover it and give a standard set of tools for web user maintenance.

If there was a CSV export and import/merge option for managers/web users as part of the admin interface then that would also be useful!

**Comment by Deesen** [comment]
Then I would call these variable user-variables = UVs.. what would be the nessecary options per UV?

1. Which types of "UVs" should be available as user-fields? Input-text, checkbox, radio, select?
2. User-group checkboxes per UV to assign it?
3. Checkbox per UV to assign it to manager and/or web-users?
4. Checkbox per UV for setting it "Internal/Manager user only" (if available in snippet Weblogin or not)

CSV-export/import means exporting an existing set of selected UVs into CSV (or JSON?), to be able to import/merge these fields into a new MODX-installation?

**Comment by bossloper** [comment]
RE: CSV export/import  - yes + potentially a CSV of a minimum of username, name and email addresses would import from other (non-MODX) sources. I would also suggest 'Full name' should be split into 2: forenames and lastname as it is better for listing users by lastname alphabetically etc.

**Comment by Deesen** [comment]
Ahh sry, I was only thinking about UVs and missed that you wrote "user"-export, not "uv"-export. Flexible user-import/export sounds very interesting indeed!

**Comment by bossloper** [comment]
ah ok. Yes user export - useful for importing email addresses into a third party system like MailChimp for example. If it exported all fields (UVs) then would make a useful backup in conjunction with import/merge, or maybe the fields to export could be selected. I think there a couple of old modules out there that do something similar but if the admin interface is getting updated then is seems like a core feature and should be built in.

Anything logical to reduce the number of active modules is good in my book as that horizontal module line can get quite long, sometimes too long for a smaller screen. Though I presume BS3 manager interface will solve that!

**Comment by Breezer** [comment]
I had to create this functionality for the forum I was making for evolution, creating web user fields which could be edited in front end and also appear in the manager interface via a plugin. If anyone wants to see how it works, go and create a web user account at frsbuilders.net or grab the forum download and install it locally. I'll be updating the forum download code later this weekend with a few fixes.
Basically it takes a string which has the option name, option value, and option text and creates a setting in the database. Useful maybe, maybe not :)

**Comment by Nicola1971** [comment]
@Breezer I always found very interesting your evo forum project. I wanted to try but it seems not work with latest Evo releases.

**Comment by Breezer** [comment]
Yes there was a change in 1.1 to makeURL that kaboshed the ajax in the manager, I have it fixed locally but I haven't had time to adjust the download files ( my computer with all my work files died ugh ). I'll be doing that tomorrow for sure.  It's going to be getting cold here shortly so I'll be back indoors working on the forum again :)

**Comment by mcparana** [comment]
there is table {PREFIX}web_user_settings in evo initial sql (at lest from 1.0.5 till now), and its always empty.

maybe this feature was planned?

**Comment by dmi3yy** [comment]
move to new Evo repo
